chapters/feature-selection.qmd

```diff
@@ -0,0 +1,525 @@
+
+# ------------------------------------------------------------------------------
+
+library(heatmaply)
+library(future)
+library(tidymodels)
+library(colino)
+library(patchwork)
+
+# ------------------------------------------------------------------------------
+# Set options
+
+tidymodels_prefer()
+theme_set(theme_transparent())
+plan("multisession")
+set_options()
+
+# ------------------------------------------------------------------------------
+# load data and results
+
+source("../R/setup_deliveries.R")
+load("../RData/deliveries_cubist.RData")
+load("../RData/deliveries_lm.RData")
+load("../RData/forested_data.RData")
+
+ctrl_rs <- control_resamples(
+  save_pred = TRUE,
+  parallel_over = "everything",
+  save_workflow = TRUE
+)
+
+reg_metrics <- metric_set(mae, rmse, rsq)
+```
+
+In Chapters [-@sec-numeric-predictors] through [-@sec-interactions-nonlinear], we described different methods for adding new predictors to the model (e.g., interactions, spline columns, etc.). We can add any predictors (or functions of predictors) based on first principles or our intuition or contextual knowledge. 
+
+We are also motivated to remove unwanted or irrelevant predictors. This can decrease model complexity, training time, computer memory, the cost of acquiring unimportant predictors, and so on. For some models, the presence of non-informative predictors can _decrease_ performance. 
+
+We may want to remove predictors based on their distributions. Unsupervised feature selection focuses on just the predictor data. 
+
+More often, though, we are interested in supervised feature selection^[Recall that we use the terms "predictors" and "features" interchangeably. We favor the former term, but "features" is more commonly used when discussing the filtering of columns.]: the removal of predictors that do not appear to have a relationship to the outcome. This implies that we have a statistic that can compute some measure of each predictor’s importance. 
+
+While it is easy to add predictors to a model, it can be very difficult (and computationally expensive) to properly determine which predictors to remove. The reason comes back to the misuse (or over-use) of data. We’ve described the problem of using the same data to fit and evaluate a model. By naively re-predicting the training set, we can accidentally get overly optimistic performance statistics. To compensate for this, resampling repeatedly allocates some data for model fitting and other data for model evaluation. 
+
+For feature selection, we have the same problem when the same data are used to measure importance, filter the predictors, and then train the model on the smaller feature set. It is similar to repeatedly taking the same test and then measuring our knowledge using the grade for the last test repetition. It doesn’t accurately measure the quantity that is important to us. This is the reason that this chapter is situated here instead of in Part 2; understanding overfitting is crucial. 
```

Comment:
> This is the reason that this chapter is situated here instead of in Part 2; **understanding overfitting is crucial.**

this might be an artifact of me reading out of order, but this last sub-sentence feels like it comes out of nowhere